compiler/rustc_lint/src/early.rs

```diff
@@ -424,10 +424,25 @@ pub fn check_ast_node<'a>(
     let mut passes: Vec<_> = passes.iter().map(|p| (p)()).collect();
     let mut buffered = lint_buffer.unwrap_or_default();
 
-    if !sess.opts.debugging_opts.no_interleave_lints {
+    if sess.opts.debugging_opts.no_interleave_lints {
+        for (i, pass) in passes.iter_mut().enumerate() {
+            buffered =
+                sess.prof.extra_verbose_generic_activity("run_lint", pass.name()).run(|| {
+                    early_lint_node(
+                        sess,
+                        !pre_expansion && i == 0,
```

Comment:
```suggestion
                        // We want to only warn about weird lints
                        // once. As it doesn't matter when we lint,
                        // we do it as part of the first pass.
                        !pre_expansion && i == 0,
```
That's my understanding here, is that correct?